Mathlib/Data/Matrix/Notation.lean

```diff
@@ -11,157 +11,129 @@ Authors: Anne Baanen, Eric Wieser
 import Mathlib.Data.Matrix.Basic
 import Mathlib.Data.Fin.VecNotation
 import Mathlib.Tactic.FinCases
+import Mathlib.Tactic.ToExpr
 import Mathlib.Algebra.BigOperators.Fin
+import Qq
 
 /-!
 # Matrix and vector notation
 
-This file includes `simp` lemmas for applying operations in `data.matrix.basic` to values built out
-of the matrix notation `![a, b] = vec_cons a (vec_cons b vec_empty)` defined in
-`data.fin.vec_notation`.
+This file includes `simp` lemmas for applying operations in `Data.Matrix.Basic` to values built out
+of the matrix notation `![a, b] = vecCons a (vecCons b vecEmpty)` defined in
+`Data.Fin.VecNotation`.
 
-This also provides the new notation `!![a, b; c, d] = matrix.of ![![a, b], ![c, d]]`.
-This notation also works for empty matrices; `!![,,,] : matrix (fin 0) (fin 3)` and
-`!![;;;] : matrix (fin 3) (fin 0)`.
+This also provides the new notation `!![a, b; c, d] = Matrix.of ![![a, b], ![c, d]]`.
+This notation also works for empty matrices; `!![,,,] : Matrix (Fin 0) (Fin 3)` and
+`!![;;;] : Matrix (Fin 3) (Fin 0)`.
 
 ## Implementation notes
 
-The `simp` lemmas require that one of the arguments is of the form `vec_cons _ _`.
+The `simp` lemmas require that one of the arguments is of the form `vecCons _ _`.
 This ensures `simp` works with entries only when (some) entries are already given.
 In other words, this notation will only appear in the output of `simp` if it
 already appears in the input.
 
 ## Notations
 
 This file provide notation `!![a, b; c, d]` for matrices, which corresponds to
-`matrix.of ![![a, b], ![c, d]]`.
-A parser for `a, b; c, d`-style strings is provided as `matrix.entry_parser`, while
-`matrix.notation` provides the hook for the `!!` notation.
-Note that in lean 3 the pretty-printer will not show `!!` notation, instead showing the version
+`Matrix.of ![![a, b], ![c, d]]`.
+Note that in lean 4 the pretty-printer will not show `!!` notation, instead showing the version
```

Comment:
We should add a TODO here to implement the unexpander; we didn't have an option in Lean 3, hence the original comment.